File: frontend/src/core/models/event.js
export class Event {
  constructor({
    id,
    title,
    description = "",
    location = "",
    start,
    end,
    personIds = [],
    frequency = "none",
    interval = 1,
    exclusions = [],
  } = {}) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.location = location;
    this.start = start instanceof Date ? start : new Date(start);
    this.end = end instanceof Date ? end : new Date(end);
    this.personIds = Array.isArray(personIds) ? personIds.slice() : [];
    this.frequency = frequency;
    this.interval = interval || 1;
    this.exclusions = Array.isArray(exclusions) ? exclusions.slice() : [];
  }
}

File: backend/internal/model/event.go
package model

// Event mirrors the `Event` entity in docs/asyncapi.yaml and the frontend
// EventDTO. start_at/end_at are ISO 8601 / RFC 3339 strings; person_ids carries
// the assigned person ids (junction table in the DB, array on the wire).
// Interval is the recurrence interval (every N units of frequency; 1 = every).
// Exclusions is a list of occurrence start times (ISO 3339) to skip when
// expanding a recurring event — supports deleting a single occurrence without
// removing the whole series.
type Event struct {
	ID         int      `json:"id"`
	Title      string   `json:"title"`
	Description string   `json:"description"`
	Location    string   `json:"location"`
	StartAt    string   `json:"start_at"`
	EndAt      string   `json:"end_at"`
	PersonIDs  []int    `json:"person_ids"`
	Frequency  string   `json:"frequency"`
	Interval   int      `json:"interval"`
	Exclusions []string `json:"exclusions"`
}
